### PR TITLE
User and logInAction implementation done.

### PR DESCRIPTION
**Pull request by a-llison-lau**

### CODE REVIEW

**Review by Andy-Guo-21 — APPROVED** [review]
Add a comment on the overloaded constructor in the User class.
Also, tests are required.
Otherwise, looks fine to me

**Comment by enfz** [comment]
User class looks good, look to add some unit tests for the methods in the class. Error Catching to be replaced with an external window later on. GUI for the login system to be added later on. Registering a user, changing your password, and logging in can be split into new use cases.

File: src/main/java/logInAction.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        Scanner scanner1 = new Scanner(new File("users.txt"));
+        boolean contains = false;
+        while (scanner1.hasNextLine()) {
+            String line = scanner1.nextLine();
+            if (line.contains(user.getName())) {
+                contains = true;
+            }
+        }
+        if (! contains) {
+            System.out.println("Username does not exist.");
```
Comment (Jefferson-liu) [inline src/main/java/logInAction.java:75]: Going to have to change to display on gui

**Review by ericshi03 — APPROVED** [review]
User class looks good except I don’t think balance should be static, you could just set it to 100 for all user constructors. Your logInAction also should be separated into two classes - one for each use case to abide by clean architecture.

File: src/main/java/logInAction.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    /**
+     * Logs the User into the system.
+     * If the User enters the correct combination of name, password and type, the User logs in.
+     * If the User enters a name that exists in the file but wrong password, the system displays wrong password.
+     * If the User enters a name that does not exist in the file, the system creates a new User.
+     * @param name name of the User
+     * @param password password of the User
+     */
+    public static void logIn(String name, String password, String type) throws FileNotFoundException {
```
Comment (Jefferson-liu) [inline src/main/java/logInAction.java:14]: Do methods have to be static?

**Review by Jefferson-liu — APPROVED** [review]
The code looks ok for now, but we may have to change the methods to work on our GUI.

**Pull request merged**